planner/optimize.go

```diff
@@ -78,7 +78,16 @@ func matchSQLBinding(sctx sessionctx.Context, stmtNode ast.StmtNode) (bindRecord
 
 // Optimize does optimization and creates a Plan.
 // The node must be prepared first.
-func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is infoschema.InfoSchema) (plannercore.Plan, types.NameSlice, error) {
+func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is infoschema.InfoSchema) (plan plannercore.Plan, _ types.NameSlice, err error) {
+	defer func() {
```

Comment:
The `Optimize` function is to generate a logical plan, how about moving the transaction-related optimizing and warm-up logic out.

Reply:
Okay, I have moved it out.